src/pkg/packager/creator/skeleton.go

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package creator contains functions for creating Zarf packages.
+package creator
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config/lang"
+	"github.com/defenseunicorns/zarf/src/extensions/bigbang"
+	"github.com/defenseunicorns/zarf/src/internal/packager/helm"
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/mholt/archiver/v3"
+)
+
+var (
+	// veryify that skeletonCreator implements Creator
+	_ Creator = (*skeletonCreator)(nil)
+)
+
+// skeletonCreator provides methods for creating skeleton Zarf packages.
+type skeletonCreator struct {
+	createOpts  types.ZarfCreateOptions
+	publishOpts types.ZarfPublishOptions
+}
+
+// LoadPackageDefinition loads and configure a zarf.yaml file when creating and publishing a skeleton package.
+func (sc *skeletonCreator) LoadPackageDefinition(dst *layout.PackagePaths) (loadedPkg *types.ZarfPackage, warnings []string, err error) {
+	var pkg types.ZarfPackage
+
+	if err := utils.ReadYaml(layout.ZarfYAML, &pkg); err != nil {
+		return nil, nil, fmt.Errorf("unable to read the zarf.yaml file: %w", err)
+	}
+
+	configuredPkg, err := setPackageMetadata(pkg, sc.createOpts)
+	if err != nil {
+		message.Warn(err.Error())
+	}
+
+	// Compose components into a single zarf.yaml file
+	composedPkg, composeWarnings, err := ComposeComponents(configuredPkg, sc.createOpts.Flavor)
+	if err != nil {
+		return nil, nil, err
+	}
+	warnings = append(warnings, composeWarnings...)
+
+	extendedPkg, err := sc.processExtensions(composedPkg, dst)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, warning := range warnings {
+		message.Warn(warning)
+	}
+
+	loadedPkg = extendedPkg
```

Comment:
Just return extendedPkg or rename extendedPkg to loadedPkg. The named return parameter should match what you end up doing

Reply:
Talked to @Noxsios about this and agree that the different named pkg variables are weird/confusing. Going to stick a single `pkg` variable to update and return in the functions instead of having a new variable name after every function call.